src/main/java/MealDemo/service/MealPlanningService.java

```diff
@@ -19,5 +21,15 @@ public interface MealPlanningService {
 
     boolean deleteMeal(int id);
 
+    Map<String, List<Meal>> mealsByCategories();
```

Comment:
Be consistent with return types in your interfaces. Typically it's good practice to return the most applicably generic Datatype. In this case return `Maps` rather than `HashMaps` since `Map` is a parent class of `HashMap` it will just work. No need to make implementation detail changes (i.e. the code inside the method)